utils/src/jsMain/kotlin/com/badoo/reaktive/utils/Timers.kt

```diff
@@ -0,0 +1,42 @@
+package com.badoo.reaktive.utils
+
+import kotlin.browser.window
+
+external fun setTimeout(handler: dynamic, timeout: Int = definedExternally, vararg arguments: Any?): Int
+external fun setInterval(handler: dynamic, timeout: Int = definedExternally, vararg arguments: Any?): Int
+external fun clearTimeout(handle: Int = definedExternally)
+external fun clearInterval(handle: Int = definedExternally)
+
+object Timers {
+    fun setTimeout(delayMillis: Long, task: () -> Unit, vararg arguments: Any?): Int {
+        return try {
+            window.setTimeout(task, delayMillis.toInt(), arguments)
+        } catch (ex: dynamic) {
```

Comment:
Actually that was my first approach in this commit https://github.com/badoo/Reaktive/pull/486/commits/c8c797ba7c7c31f28acaf6248582f6587ae230f5
But then if you define`window` as a workaround 
`global.window = {document: {createElementNS: () => {return {}} }};`
Or for some reason window is defined (not sure it's possible or not)
Then this fix will be useless because window will be defined so you will get the error 
`TypeError: window.setTimeout is not a function`

What do you think is better then? Should check if window is defined and try catch it just in case we get the other error?

Reply:
Can you try `if (window == undefined) ... else ...` ?

Reply:
maybe we can use external setTimeout function in both browser and NodeJS environment?

Reply:
@lukaville I did not try since I am not testing against browser, but if you believe it will work for both, then I don't mind just directly calling it. What do you think?

Reply:
@arkivanov in my case `window` was not `undefined` which cause the `setTimeout` is not a function error. So window can be undefined and still get the error in nodejs

Reply:
Can you check somehow whether `window.setTimeout` is defined our not? `try-catch` is really a bad solution and we should try to find another way.

Reply:
I could not do that, I just put window check for now. If calling setTimeout will work for browser and NodeJS then maybe just use it and ignore window

Reply:
The `window` should not be defined in NodeJs. There some related explanations [here](https://stackoverflow.com/questions/45964178/referenceerror-window-is-not-defined-at-object-anonymous-node-js). I will try to find a way of how to check this in kotlin. The pure JS way is as follows: `if (typeof window !== 'undefined')`.

Reply:
Maybe this could somehow work? https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.js/js-type-of.html

Reply:
Also this may help: https://kotlinlang.org/docs/reference/js-interop.html#inline-javascript